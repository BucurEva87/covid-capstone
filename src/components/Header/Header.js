import React from 'react';
import { Link } from 'react-router-dom';
import style from './Header.module.scss';

const Header = () => (
  <header data-testid="header">
    <span className={style.back}><Link to="/">&laquo;</Link></span>
    <span>Statistics</span>
    <div>
      <img src={`${process.env.PUBLIC_URL}images/microphone.png`} alt="Microphone" />
      <img src={`${process.env.PUBLIC_URL}images/settings.png`} alt="Settings" />
    </div>
  </header>
);

export default Header;
