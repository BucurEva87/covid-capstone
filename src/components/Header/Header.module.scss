@import '../../styles/variables';

header {
  width: 100%;
  height: 10vh;
  color: $linkColor;
  display: flex;
  align-items: center;
  font-size: 5rem;
  justify-content: space-around;
}
