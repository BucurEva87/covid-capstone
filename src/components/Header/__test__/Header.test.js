import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '../Header';

let getByTestId;

beforeEach(() => {
  const component = render(
    <Router>
      <Header />
    </Router>,
  );

  getByTestId = component.getByTestId;
});

describe('Header component', () => {
  test('renders on the screen', () => {
    expect(getByTestId('header')).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const tree = renderer.create(
      <Router>
        <Header />
      </Router>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
