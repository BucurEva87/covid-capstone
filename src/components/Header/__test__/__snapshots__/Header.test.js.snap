// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
<header
  data-testid="header"
>
  <span
    className="back"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Â«
    </a>
  </span>
  <span>
    Statistics
  </span>
  <div>
    <img
      alt="Microphone"
      src="images/microphone.png"
    />
    <img
      alt="Settings"
      src="images/settings.png"
    />
  </div>
</header>
`;
